### PR TITLE
Notify users for school email address (.edu | .ac | .sch)

### LINKED ISSUE #4393: Notify users if they are trying to use an .edu email account
Educational email addresses very often don't allow for receiving emails from outside the school network. Notify the user that they might not receive emails from monkeytype if they continue

### PR DESCRIPTION
**Pull request by aviiciii**
### Description
This pull request adds a new feature to the sign-up process that checks if the provided email address is a school email address (.edu domain). It also includes an indicator to notify users to consider using a personal email address for better email functionality.

### Demo

https://github.com/monkeytypegame/monkeytype/assets/88141521/2e2b3ab2-6e26-41fd-8758-8d1e05adca5e

Closes #4393

### CODE REVIEW

**Comment by aviiciii** [comment]
Hey, I added university domain (.ac) and school domains (.sch). 
note that .ac is a tld and a second-level domain whereas .sch is second-level domain only.

File: frontend/src/ts/pages/login.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export function hidePreloader(): void {
 export const updateSignupButton = (): void => {
   if (
     nameIndicator.get() !== "available" ||
-    emailIndicator.get() !== "valid" ||
+    (emailIndicator.get() !== "valid" && emailIndicator.get() !== "edu") ||
```
Comment (Miodec) [inline frontend/src/ts/pages/login.ts:44]: Remove. I think we should still allow the signup if the user wants to use that email

**Review by Miodec — CHANGES_REQUESTED** [review]

File: frontend/src/ts/pages/login.ts (hunk 1 above)
Comment (aviiciii) [reply frontend/src/ts/pages/login.ts:44]: (emailIndicator.get() !== "valid" && emailIndicator.get() !== "edu") 
this simply says that it must either be "valid" or "edu".

Comment (aviiciii) [reply frontend/src/ts/pages/login.ts:44]: @Miodec ?

**Comment by Miodec** [comment]
Thanks

File: frontend/src/ts/pages/login.ts (hunk 1 above)
Comment (Miodec) [reply frontend/src/ts/pages/login.ts:44]: Yeah youre right

**Pull request merged**